partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rex/ExprUpvalue.kt

```diff
@@ -0,0 +1,22 @@
+package org.partiql.eval.internal.operator.rex
+
+import org.partiql.eval.internal.Record
+import org.partiql.eval.internal.operator.Operator
+import org.partiql.value.PartiQLValue
+import org.partiql.value.PartiQLValueExperimental
+import java.util.Stack
+
+/**
+ * Returns the appropriate value from the stack.
+ */
+internal class ExprUpvalue(
```

Comment:
Perhaps we keep implementation names consistent with plan node names

- ExprVarGlobal
- ExprVarLocal
- ExprVarOuter

I'm only slightly hesitant on the upvalue name because I believe there is some lifetime implication, whereas we are more similar to the nested loop situation

```
for i in [0..n] {
   for j in [0...m] {
       // stack [ i, j ] <- top
   }
}
```

For us, we aren't concerned about lifetimes. In our case, we are calling "i" an upvalue when really it is just earlier in the traditional stack. The "upvalue" from Lua is more about keeping a value (that was previously on the stack) alive when its scope no longer exists (ie we returned, but had a closure so need to keep "i" alive). 

Our situation is exactly

```
for i in [0..n] {
   // record here [ i ]
   for j in [0...m] {
       // record here [ j ]   <-- record may have several values, this is what we call the "stack"
       // we do not concatenate records to form a singular stack, instead we do a stack-of-records aka stack-of-stacks, to create our scope
       //  scope [[i], [j]]
       //     outer^    ^local
   }
}
```

Reply:
Agreed. Will update!